### PR TITLE
Added CustomParams

### PR DESCRIPTION
**Pull request by leonfagan71**

### CODE REVIEW

**Comment by leonfagan71** [comment]
Should not have edited entire file now.
Added {message}, {match} and {player} to most action output types.

File: src/main/java/com/minecraftdimensions/bungeechatfilter/Rule.java [hunk 1]
```diff
@@ -43,81 +41,108 @@ public String getStringRegex() {
         return regex.pattern();
     }
 
-    public Matcher getMatcher( String message ) {
-        return regex.matcher( message );
+    public Matcher getMatcher(String message) {
+        return regex.matcher(message);
```
Comment (zedwick) [inline src/main/java/com/minecraftdimensions/bungeechatfilter/Rule.java:45]: Why did you modify these lines?

File: src/main/java/com/minecraftdimensions/bungeechatfilter/Rule.java [hunk 2]
```diff
@@ -43,81 +41,108 @@ public String getStringRegex() {
...
     }
 
-    public boolean doesMessageContainRegex( String message ) {
-        return getMatcher( message ).find();
+    public boolean doesMessageContainRegex(String message) {
+        return getMatcher(message).find();
```
Comment (zedwick) [inline src/main/java/com/minecraftdimensions/bungeechatfilter/Rule.java:49]: Another example of lines modified to remove spaces, with not code changes.

File: src/main/java/com/minecraftdimensions/bungeechatfilter/Rule.java [hunk 3]
```diff
@@ -43,81 +41,108 @@ public String getStringRegex() {
...
-            } else if ( action.equals( "message" ) ) {
-                player.sendMessage( TextComponent.fromLegacyText( Main.color( actions.get( action )[0] ) ) );
-            } else if ( action.equals( "kick" ) ) {
-                player.disconnect( new TextComponent( TextComponent.fromLegacyText( Main.color( actions.get( action )[0] ) ) ) );
-            } else if ( action.equals( "alert" ) ) {
-                String alert =   actions.get( action )[0].replace( "{player}", player.getDisplayName() );
-                if(message.split( " ", 2 ).length>1){
-                       alert =alert.replace("{arguments}", message.split( " ", 2 )[1] )    ;
+        for (String action : actions.keySet()) {
+            if (action.equals("deny")) {
```
Comment (zedwick) [inline src/main/java/com/minecraftdimensions/bungeechatfilter/Rule.java:61]: I can see that these lines at the top have just been reformatted again, which makes it difficult to easily read the actual changes made in this block of changed lines (and we don't want you reformatting the code in this PR)

**Review by zedwick — CHANGES_REQUESTED** [review]
You've still removed a ton of spaces in lines you didn't need to modify.

File: src/main/java/com/minecraftdimensions/bungeechatfilter/Rule.java (hunk 3 above)
Comment (zedwick) [reply src/main/java/com/minecraftdimensions/bungeechatfilter/Rule.java:61]: I did not comment on all the lines you've reformatted, please try to ensure your PR is only changing the lines needed for your feature contribution.

**Comment by leonfagan71** [comment]
I essentially just created a new pull request and coped the files I was working on before into the directory.
you asked me to remove spaces in the last pull request so I ran it through a formatter, it fixed the indentation issues that made it harder to understand. it was then filled with tabs instead of spaces so I replaced them.

At this point there was a ton of code changes so I closed the PR and created this one.
I don't really have enough time to keep making changes to make it look worse than before.
Extra spaces and new lines just look terrible, yes it can sometimes help people to see what's going on but for most of these functions, you don't really need to have them.
New line 83 is the only example where it's harder to work out what's going on with that line.

**Comment by zedwick** [comment]
Making frivolous formatting changes to a file by running it blindly through a formatter out of laziness just to avoid changing 2/3 lines by hand is not an ideal way to contribute code to a project, as stated.

Thank you for taking the time to make these PRs. I'll likely come back to read through them at some point when I come to implement the feature.

**Pull request closed without merging**